File: Core CS/Algorithms/Implementation/Bill Division/billDivision.java
import java.io.*;
import java.util.*;
import java.text.*;

public class Solution {

    static void bonAppetit(int k, int b, int[] ar) {
        int s = 0 - ar[k];
        for(int a: ar)
            s += a;
        
        if(b - s/2 == 0)
            System.out.println("Bon Appetit");
        else
             System.out.println(b - s/2);
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int k = in.nextInt();
        int[] ar = new int[n];
        for(int ar_i = 0; ar_i < n; ar_i++){
            ar[ar_i] = in.nextInt();
        }
        int b = in.nextInt();
        bonAppetit(k, b, ar);
    }
}

File: Core CS/Algorithms/Implementation/Bill Division/billDivision.cpp
#include <bits/stdc++.h>
using namespace std;

void bonAppetit(int k, int b, vector <int> ar) {
    int s = 0 - ar[k];
    for(int a: ar)
        s += a;
        
    if(b - s/2 == 0)
        cout << "Bon Appetit";
    else
        cout << b - s/2;
}

int main() {
    int n;
    int k;
    cin >> n >> k;
    vector<int> ar(n);
    for(int ar_i = 0; ar_i < n; ar_i++){
       cin >> ar[ar_i];
    }
    int b;
    cin >> b;
    bonAppetit(k, b, ar);
}
